.github/workflows/build_SRWA.yml

```diff
@@ -94,15 +94,32 @@ jobs:
     runs-on: self-hosted 
     if: ${{ github.event.label.name == 'ci-aws-intel-build' }}
     needs: [ Set_Repo_and_Branch, Clone_SRWA, Configure_SRWA, Build_SRWA ]
-    
+    outputs: 
+      build-success: ${{ steps.status.outputs.build-success }}
+
     steps:
       - name: Get build status
+        id: status
         run: |
           cd ufs-srweather-app-${{ github.run_id }}/test/
           status=`tail -1 build_test.out |cut -d ' ' -f 3-`
           if [ $status == PASS ]; then
+            echo "::set-output name=build-success::success"
             exit 0
           else 
+            echo "::set-output name=build-success::failure"
             exit 1
           fi
 
+  capture_workflow_run-id:
+    name: Save workflow_run id of SRWA build 
+    needs: Build_Status
+    if: ${{ github.event.label.name == 'ci-aws-intel-build' && needs.Build_Status.outputs.build-success == 'success' }}
+    runs-on: self-hosted
+    steps: 
+      - uses: actions/checkout@v3
+        with:
+          clean: false
+      - name: Save SRWA run_id  
+        run: |
+          echo ${{ github.run_id }} > ${{ github.workspace}}/run_id.txt
```

Comment:
Is this workspace common to all the tests that are run? If so, this doesn't seem safe if running 2 PR tests simultaneously. I'm not sure how you're using this downstream, but it could be effective to touch a file with the run_id in its name so you avoid having multiple instantiations overwrite this file.

Reply:
> Is this workspace common to all the tests that are run?

Yes.

> If so, this doesn't seem safe if running 2 PR tests simultaneously.

 You're correct, it isn't "safe" the way it currently stands. The way that is being accounted for is that there is a level of human intervention in attaching the labels to PRs, which should only be done one at a time for now. 

It gets complicated in getting access to a previous workflow's `run_id` for use in downstream workflows (WE2E Tests).  The `run_id` of a previous workflow isn't easily obtainable unless the previous workflow (Build_SRWA) triggered the current workflow (WE2E Tests).  We can do that by specifying the `on: workflow_run:` trigger, which is the way I previously structured the WE2E workflow. (ie: run the WE2E workflow only on successful completion of the build_SRWA workflow).  This couples the two workflows, limiting the ability to run the WE2E tests in isolation. 

I thought about using the PR number instead of the `run_id` as the file name, but the problem is the "event" that triggers the workflow is attaching a label, not the PR, so we can't set the `github.event.number` to the PR number. Also, if one were to manually trigger the workflow, the PR number wouldn't be available either. 

In the case where a code owner only wants to run the WE2E tests (NOT build the app), they would not be able to do that in attaching the "ci-aws-intel-WE" label. Triggering the WE2E Test workflow by solely attaching the "ci-aws-intel-WE" label doesn't have access to the `run_id` of the build workflow. 

Long explanation here, but I've tried to strike a balance all of the above - accounting for multiple PRs, - building the SRWA every time, - how the workflow is triggered (label vs every PR).  I sided on being more judicious with compute resources at the expense of the code owner being aware of the limitation of only being able to run the test suite for one PR at a time.  

If there is a strong opinion on implementing this a certain way, please let me know and I can make the adjustments.

Reply:
@robgonzalezpita / @christinaholtNOAA - I did some looking around to see how to obtain the PR# and other information inside a GA, and you are right that it is not straightforward at all.  There are different approaches people have found to work around the problem, but none of them seem to be totally reliable.  And one of the main issues, as you point out, is the availability of the information when the GA is initiated by different events.  The strategy for getting that info is dependent on the particular event that triggered the GA.  My inclination is to not worry about this at the moment because it's still an open question in the Github community how to get this information reliably.

One thing we could potentially do is have the build job create a UUID of some kind, and just use that to uniquely identify each invokation of the GA.